File: app/models/commentary.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.sql import func
from ..database import Base

class Commentary(Base):
    __tablename__ = "t_app_commentary"
    
    id = Column(Integer, primary_key=True, index=True)
    comment = Column(String(256), nullable=False)
    user_id = Column(String(100), nullable=False)
    product_id = Column(Integer, nullable=False)
    
    def __str__(self):
        return f'Commentary {self.id}' 

File: models/Commentary.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');
const User = require('./User');
const Product = require('./Product');

const Commentary = sequelize.define('Commentary', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  content: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'auth_user',
      key: 'id'
    }
  },
  product_id: {
    type: DataTypes.BIGINT,
    allowNull: false,
    references: {
      model: 't_app_product_product',
      key: 'id'
    }
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'commentaries',
  timestamps: false
});

// Associations
Commentary.belongsTo(User, { foreignKey: 'user_id', as: 'user' });
Commentary.belongsTo(Product, { foreignKey: 'product_id', as: 'product' });
User.hasMany(Commentary, { foreignKey: 'user_id', as: 'commentaries' });
Product.hasMany(Commentary, { foreignKey: 'product_id', as: 'commentaries' });

module.exports = Commentary; 